### ISSUE TITLE
Re-imagining the typing experience on a smartphone

### ISSUE THREAD

**Issue by sawtdakhili** [open-source-ideas/ideas#364]
## Project description

**Problem:** 
It’s really inconvenient to write long-form content on a smartphone. When the phone is in portrait mode, there’s enough room to see all the text, but the keyboard is too tiny for extended writing sessions resulting in fatigue. When you turn the smartphone to landscape mode, two problems arise: the keyboard takes up two-thirds of the screen, making it hard to see what you’re typing, and reaching the keys in the middle of the keyboard becomes difficult. Thumb mode isn’t a satisfactory solution either, as it leaves a sad stupid wasted space in the middle of the split keyboard.

When writing on a smartphone, screen real estate is precious. Tablets offer a split thumb floating keyboard mode, but they are bulky, heavy, and not comfortable for long-term typing. Carrying a tablet everywhere isn’t practical.

**What’s available:** 
I’ve tried a dozen writing apps (even Joe’s Termux with its high-condensed text) combined with several Android keyboards, but none come close to the experience of having a real keyboard (I prefer ortholinear split keyboards btw) attached to a smartphone for travel.

So, how can we fix the typing experience on a phone?

**Solution:** 
The ideal solution would be to have a split keyboard that occupies the two top sides of the screen, with the text editor displayed in the middle. Full-screen mode to get rid of the status bar would be a plus. You would hold your phone like a Steam Deck or, even better, like the Nokia N-Gage (I know that there have been phones closer to what I’m describing, but the N-Gage is the most well-known form factor).

![apple-mail-layout](https://github.com/open-source-ideas/ideas/assets/70063536/c5a31ca1-290e-416d-b233-3e7786fb28f5)
![long-form-writing-layout](https://github.com/open-source-ideas/ideas/assets/70063536/fe50d338-acb6-426f-86c0-ba4b759608aa)
![obsidian-layout](https://github.com/open-source-ideas/ideas/assets/70063536/9dd2508a-d1e9-4f0f-9b56-f00de5fdb2f9)
![termux-mockup](https://github.com/open-source-ideas/ideas/assets/70063536/d7232e11-230f-466e-abc2-bc424a2dca59)
![long-form-writing](https://github.com/open-source-ideas/ideas/assets/70063536/504aeace-87a0-4ec0-a06f-9f9460e4fbc3)
![gmail](https://github.com/open-source-ideas/ideas/assets/70063536/bea17913-f793-48f1-857d-d8a80176bc60)
![obsidian](https://github.com/open-source-ideas/ideas/assets/70063536/7f19b968-8641-4c3c-a646-b3eaee349dfc)
![apple-mail](https://github.com/open-source-ideas/ideas/assets/70063536/20a8c3f1-2b83-414b-ac9f-66ce8d8ba6be)
![termux](https://github.com/open-source-ideas/ideas/assets/70063536/602e8e5f-2099-43e6-8ea0-513b93d7ed41)

**Additional benefits:** 
This layout would be more comfortable to hold, similar to how you hold your phone when playing PUBG. Your index fingers would rest on the phone, and the bottom corners of the phone would fit into the palms’ cavity. Achieving this would be possible since the keyboard would be positioned closer toward the top corners of the phone (remember, the phone is in landscape mode).

**Some other thoughts:** 
Perhaps combining a keyboard with a text editor is excessive. It’s possible that a new type of keyboard that splits and allows you to view the text you’re typing in the middle would suffice. However, it’s uncertain whether this would work with every app.

_Disclaimer: 
Please note that the mockups provided are intended solely for illustrative purposes to convey the concept and potential user experience._

## Relevant Technology

I am no programmer. I am mainly a writer tech poweruser. I pitched this idea on many places and it was quite well received.

### Complexity

- [ ] Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- [x] Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- [ ] Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- [ ] Little work - A couple of days
- [x] Medium work - A week or two
- [ ] Much work - The project will take more than a couple of weeks and serious planning is required

### Categories

- [x] Mobile app
- [ ] IoT
- [ ] Web app
- [x] Frontend/UI
- [ ] AI/ML
- [ ] APIs/Backend
- [ ] Voice Assistant
- [ ] Developer Tooling
- [ ] Extension/Plugin/Add-On
- [x] Design/UX
- [ ] AR/VR
- [ ] Bots
- [ ] Security
- [ ] Blockchain
- [ ] Futuristic Tech/Something Unique

**Comment by KOLANICH** [comment]
The ideal solution is not to use a screen space for a keyboard at all and use a hardware keyboard. Ideally the one mechanically coupled with the smartphone. A modular design - a unified frame with electronics and hardware + a plug matching the shape of a phone to the shape of the keyboard. But if we continue thinking this way ... it will lead us to the discontinued https://en.wikipedia.org/wiki/Project_Ara .

And even better solution is just using a PC. Alternatively - a BCI, but present-day BCIs are very immature and will likely be slow and inconvenient to type and battery draining.

About the keyboards on top, it will cause thumbs to stretch over the whole screen. IMHO it is inconvenient, I guess instead it should be at bottom.

And one more idea. A fully transparent keyboard, positions of which keys within the layout are just remembered.

**Comment by sawtdakhili** [comment]
> BTW, is the concept art by you? How did you make it?

Yes I made the mockups my self just to illustrate the concept.

> The ideal solution is not to use a screen space for a keyboard at all and use a hardware keyboard. 

This idea is not ment to replace physical keyboards as typing on real ones are far superior. I was just looking for a way to make typing on a phone less uncomfortable when no real keyboard is available.

> About the keyboards on top, it will cause thumbs to stretch over the whole screen. IMHO it is inconvenient, I guess instead it should be at bottom.

Probably. I was trying to mimic how mobile games hold their phone and get to play for hours without fatigue.

**Comment by sygint** [comment]
There are closed source concepts like this (gboard), but an open source one would be really cool for sure.

**Comment by vitrvm** [comment]
Really cool idea. What's the status of this project?

**Comment by sawtdakhili** [comment]
Nobody is working on it afaik.

**Comment by KaKi87** [comment]
> There are closed source concepts like this (gboard)

How so ?

All I've seen is the one-hand mode, which isn't like this, and is available on open source keyboards.

**Comment by sawtdakhili** [comment]
> All I've seen is the one-hand mode, which isn't like this, and is available on open source keyboards.

Care to share name or link?

**Comment by KaKi87** [comment]
Yep : [FUTO Keyboard](https://github.com/futo-org/android-keyboard) (open source but company-owned and using their own F-Droid repo).

There's also [FlorisBoard](https://github.com/florisboard/florisboard) (open source, community owned and available on the official F-Droid repo, but lacks important features for daily-driving, such as autocompletion, autocorrect and swipe, development is slower).
